mllib/src/test/scala/org/apache/spark/ml/regression/RandomForestRegressorSuite.scala

```diff
@@ -105,6 +108,55 @@ class RandomForestRegressorSuite extends SparkFunSuite with MLlibTestSparkContex
       }
   }
 
+  test("Random Forest variance") {
```

Comment:
The first test is meant to pass for all impurities, since it compares the variance of a forest with one tree (with bootstrapping set off). You are right, that we have to be deterministic about checking the predicted variances. I have done it for the DecisionTrees here (https://github.com/apache/spark/pull/13981) but not sure it is straightforward for a RandomForest....